src/components/icon/icon.jsx

```diff
@@ -11,18 +11,26 @@ export default function Icon(props) {
     props.currentColor === 'skip' ? '' : styles[`color-${props.currentColor}`],
   ].filter(Boolean);
 
+  const tabindex = props.active && props.role === 'button' ? 0 : -1;
+  const onKeyDown = (event) => {
+    if (event.key === 'Enter' || event.key === 'Space') {
+      props.onEnter();
+    }
+  };
+
   return (
-    <svg
-      aria-label={props.ariaLabel}
-      aria-labelledby={props.ariaLabelledBy}
-      className={classes.join(' ')}
-      id={props.id}
-      onClick={props.onClick}
-      role={props.role}
-    >
-      { props.title && <title>{props.title}</title> }
-      <use xlinkHref={`#${props.name}`} />
-    </svg>
+      <svg
+        tabIndex={tabindex}
+        aria-label={props.ariaLabel}
+        aria-labelledby={props.ariaLabelledBy}
+        className={classes.join(' ')}
+        id={props.id}
+        onClick={props.onClick}
+        role={props.role}
+      >
+        { props.title && <title>{props.title}</title> }
+        <use onKeyDown={onKeyDown} xlinkHref={`#${props.name}`} />
```

Comment:
Perhaps we should only attach the `onKeyDown` handler if `props.role === 'button'` to keep it semantic. If we do make this change, the test should change along with it (and probably a new test should exist to cover the new behavior).